source/falaise/snemo/datamodels/sim_digi_data.cc

```diff
@@ -5,73 +5,131 @@
 
 namespace snemo {
 
-namespace datamodel {
+  namespace datamodel {
 
-// Serial tag :
-DATATOOLS_SERIALIZATION_SERIAL_TAG_IMPLEMENTATION(sim_digi_data, "snemo::datamodel::sim_digi_data")
+    // Serial tag :
+    DATATOOLS_SERIALIZATION_SERIAL_TAG_IMPLEMENTATION(sim_digi_data, "snemo::datamodel::sim_digi_data")
 
-sim_digi_data::sim_digi_data() {
-  _digi_events_.clear();
-  return;
-}
-
-sim_digi_data::~sim_digi_data() { return; }
+    sim_digi_data::sim_digi_data()
+    {
+      _calo_digi_hits_.clear();
+      _tracker_digi_hits_.clear();
+      _trigger_data_.clear();
+      _auxiliaries_.clear();
+      return;
+    }
 
-bool sim_digi_data::is_valid() const { return true; }
+    sim_digi_data::~sim_digi_data()
+    {
+      return;
+    }
 
-void sim_digi_data::reset() {
-  _digi_events_.clear();
-  _auxiliaries_.clear();
-  return;
-}
+    bool sim_digi_data::is_valid() const
+    {
+      return true;
+    }
 
-const sim_digi_data::sim_digi_event_collection_type& sim_digi_data::get_digi_events() const {
-  return _digi_events_;
-}
+    void sim_digi_data::reset()
+    {
+      _calo_digi_hits_.clear();
+      _tracker_digi_hits_.clear();
+      _trigger_data_.clear();
+      _auxiliaries_.clear();
+      return;
+    }
```

Comment:
What's the rationale (use cases) for resetting the data? Should data, once created, be read-only?